src/features/hud/components/Menu.tsx

```diff
@@ -197,6 +245,70 @@ export const Menu = () => {
         </div>
       </OuterPanel>
 
+      <Modal
```

Comment:
I would suggest separating this into it's own component `./Share.tsx` - This will make it more focussed and easier to maintain

Reply:
Yep. I was thinking to do that in the beginning itself. Will do the changes.